File: app/src/main/java/com/varunbarad/popularmovies/model/data/Genre.java
package com.varunbarad.popularmovies.model.data;

import com.google.gson.Gson;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Creator: vbarad
 * Date: 2016-12-07
 * Project: PopularMovies
 */

public final class Genre {
  @Expose
  @SerializedName("id")
  private long id;
  @Expose
  @SerializedName("name")
  private String name;

  /**
   * No args constructor for serialization
   */
  public Genre() {

  }

  /**
   * @param id
   * @param name
   */
  public Genre(final long id, final String name) {
    this.id = id;
    this.name = name;
  }

  /**
   * @return The unique id for this Genre
   */
  public final long getId() {
    return id;
  }

  /**
   * @return The name for this Genre
   */
  public final String getName() {
    return name;
  }

  @Override
  public String toString() {
    return (new Gson()).toJson(this);
  }
}

File: app/src/main/java/com/varunbarad/popularmovies/model/Genre.kt
package com.varunbarad.popularmovies.model

import com.varunbarad.popularmovies.external_services.movie_db_api.models.ApiGenre

/**
 * Creator: Varun Barad
 * Date: 2019-06-05
 * Project: PopularMovies
 */
data class Genre(
    val id: Long,
    val name: String
)

fun ApiGenre.toGenre(): Genre {
    return Genre(
        id = this.id,
        name = this.name
    )
}
